### PR TITLE
fix some typos in apiserver filters

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

fix some typos

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/assign @sttts 

reopen a pr for typos fix of closed #58789.

File: staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func WithWaitGroup(handler http.Handler, requestContextMapper apirequest.Request
 
 		if !longRunning(req, requestInfo) {
 			if err := wg.Add(1); err != nil {
-				http.Error(w, "Apisever is shutting down.", http.StatusInternalServerError)
+				http.Error(w, "APIServer is shutting down.", http.StatusInternalServerError)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go]: du we have a unified spelling for this?

File: staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ import (
 func WithPanicRecovery(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer runtime.HandleCrash(func(err interface{}) {
-			http.Error(w, "This request caused apisever to panic. Look in log for details.", http.StatusInternalServerError)
+			http.Error(w, "This request caused apiserver to panic. Look in log for details.", http.StatusInternalServerError)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go]: here it is `apiserver`, not `APIServer`. I tend to the former.

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go]: Also: Look in the logs for details.

File: staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go (hunk 1 above)
Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go]: I can find `apiserver` and `APIServer`, as you say the former looks better.

File: staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go (hunk 2 above)
Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go]: ok

**Pull request merged**